modules/account-lib/src/coin/sol/utils.ts

```diff
@@ -73,10 +92,163 @@ export function isValidAmount(amount: string): boolean {
+ *
+ * @param {string} rawTransaction - transaction in base64 string format
+ * @returns {boolean} - the validation result
+ */
+export function isValidRawTransaction(rawTransaction: string): boolean {
+  try {
+    const tx = SolTransaction.from(Buffer.from(rawTransaction, 'base64'));
+    tx.serialize({ requireAllSignatures: false });
+    return true;
+  } catch (e) {
+    return false;
+  }
+}
+
+/**
+ * Verifies if signature for message is valid.
+ *
+ * @param {Buffer} serializedTx - tx as a base64 string
+ * @param {string} signature - signature as a string
+ * @param {string} publicKey - public key as base 58
+ * @returns {Boolean} true if signature is valid.
+ */
+export function verifySignature(serializedTx: string, signature: string, publicKey: string): boolean {
+  if (!isValidRawTransaction(serializedTx)) {
+    throw new UtilsError('Invalid serializedTx');
+  }
+  if (!isValidPublicKey(publicKey)) {
+    throw new UtilsError('Invalid publicKey');
+  }
+  if (!isValidSignature(signature)) {
+    throw new UtilsError('Invalid signature');
+  }
+  const msg = SolTransaction.from(Buffer.from(serializedTx, 'base64')).serializeMessage();
+  const sig = base58ToUint8Array(signature);
+  const pub = new PublicKey(publicKey);
+  return nacl.sign.detached.verify(msg, sig, pub.toBuffer());
+}
+
+/**
+ * Converts a base58 string into a Uint8Array.
+ *
+ * @param {string} input - a string in base58
+ * @returns {Uint8Array} - an Uint8Array
+ */
+export function base58ToUint8Array(input: string): Uint8Array {
+  return new Uint8Array(bs58.decode(input));
+}
+
+/**
+ * Converts a Uint8Array to a base58 string.
+ *
+ * @param {Uint8Array} input - an Uint8Array
+ * @returns {string} - a string in base58
+ */
+export function Uint8ArrayTobase58(input: Uint8Array): string {
+  return bs58.encode(input);
+}
+
+/**
+ * Count the amount of signatures are not null.
+ *
+ * @param {SignaturePubkeyPair[]} signatures - an array of SignaturePubkeyPair
+ * @returns {number} - the amount of valid signatures
+ */
+export function countNotNullSignatures(signatures: SignaturePubkeyPair[]): number {
+  return signatures.filter((sig) => !!sig.signature).length;
+}
+
+/**
+ * Check if all signatures are completed.
+ *
+ * @param {SignaturePubkeyPair[]} signatures - signatures
+ * @returns {boolean}
+ */
+export function requiresAllSignatures(signatures: SignaturePubkeyPair[]): boolean {
+  return signatures.length > 0 && countNotNullSignatures(signatures) === signatures.length;
+}
+
+/**
+ * Returns the transaction Type based on the  transaction instructions.
+ * Only Wallet initialization and Transfer transactions are supported.
+ *
+ * @param {SolTransaction} transaction - the solana transaction
+ * @returns {TransactionType} - the type of transaction
+ */
+export function getTransactionType(transaction: SolTransaction): TransactionType {
+  const { instructions } = transaction;
+  validateIntructionTypes(instructions);
+  if (
+    instructions.length === 2 &&
```

Comment:
if we use a durable nonce account, we'll have 2 instructions as well:
1. advance nonce
2. transfer

Reply:
Wallet init should only have the create and initializeNonce, in the specific order mentioned.
Transfers can have advanceNonce, transfer and memo (for now)

Im limiting the intructions types to the type of tx what we want to support.